xip/xipether.c

```diff
@@ -17,13 +17,116 @@
 #include "libnetlink.h"
 #include "xiart.h"
 
+#define IFINDEX_STR_SIZE 8
+#define XIA_LLADDR_LEN 12
+#define APPEND_ZEROES 20
+
 static int usage(void)
 {
 	fprintf(stderr,
-"Usage:	\n");
+"Usage:	xip ether { addinterface | delinterface } dev DEV\n"
+"       xip ether { addneigh | delneigh } lladdr LLADDR dev DEV\n"
+"       xip ether show { interfaces | neighs }\n"
+"where	LLADDR := HEXDIGIT{1,2} (':' HEXDIGIT{1,2})*\n"
+"	DEV := STRING NUMBER\n");
 	return -1;
 }
 
+static void form_ether_xid(unsigned oif, unsigned char *lladdr, unsigned tlen, char *id)
+{
+	char bytes[4];
+	unsigned append = APPEND_ZEROES;
+	bytes[0] = (oif >> 24) & 0xFF;
+	bytes[1] = (oif >> 16) & 0xFF;
+	bytes[2] = (oif >> 8) & 0xFF;
+	bytes[3] = oif & 0xFF;
+	snprintf(id, (tlen+1), "%02x%02x%02x%02x%02x%02x%02x%02x%02x%02x%0*hu",(unsigned char)bytes[0],
+				(unsigned char)bytes[1],(unsigned char)bytes[2],
```

Comment:
Please pull back this and the following lines, since there are too many indentation. Here, we only use one more tab from the beginning of ``snprintf()``.